app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,47 @@
+class PagesController < ApplicationController
+  #before_action :set_page, except: [:index, :new, :create]  
+  before_action :set_page, only: [:show, :edit, :update, :destroy]  
+  def index
+        @pages = Page.all
+    end    
+
+    def show
+      set_page
```

Comment:
all set_page calls aren't needed when you set the "before_action" and you specify which methods you want to call that. Before_action will call that method automatically. So just keep that in mind. 👍🏼